devU-client/src/components/misc/navbar.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react'
+import withBreadcrumbs from 'react-router-breadcrumbs-hoc'
+import { Link } from 'react-router-dom'
+import styles from './navbar.scss'
+import RequestService from 'services/request.service'
+import { Assignment, Course, User } from 'devu-shared-modules'
+
+const UserBreadcrumb = ({ match }: any) => {
+    const [userName, setUserName] = useState('')
+
+    useEffect(() => {
+        getUser()
+    }, [])
+
+    const getUser = async () => { 
+        const user = await RequestService.get<User>( `/api/users/${match.params.userId}` ) 
+        setUserName(user.externalId)
+    }
+
+    return (<>{userName}</>)
+}
+
+const CourseBreadcrumb = ({ match }: any) => {
+    const [courseName, setCourseName] = useState('')
+
+    useEffect(() => {
+        getCourse()
+    }, [])
+
+    const getCourse = async () => { 
+        const course = await RequestService.get<Course>( `/api/courses/${match.params.courseId}` ) 
+        setCourseName(course.name)
+    }
+
+    return (<>{courseName}</>)
+}
+
+const AssignmentBreadcrumb = ({ match }: any) => {
+    const [assignmentName, setAssignmentName] = useState('')
+
+    useEffect(() => {
+        getAssignment()
+    }, [])
+
+    const getAssignment = async () => { 
+        const assignment = await RequestService.get<Assignment>( `/api/assignments/${match.params.assignmentId}` ) 
+        setAssignmentName(assignment.name)
+    }
+
+    return (<>{assignmentName}</>)
+}
+
+const routes = [
+    { path: '/', breadcrumb: 'Home' },
```

Comment:
For the above suggestion I have create a fix if the first route does not contain userId,

```
const DynamicBreadcrumb = ({ match }: any) => {
const pathSegment = match.url.substr(match.url.lastIndexOf('/') + 1);

const breadcrumbName = pathSegment.charAt(0).toUpperCase() + pathSegment.slice(1);

    return <>{breadcrumbName}</>;
}
const routes = [
    { path: '/', breadcrumb: 'Home' },
    { path: '/:path', breadcrumb: DynamicBreadcrumb },
    { path: '/users/:userId', breadcrumb: UserBreadcrumb},
    { path: '/courses/:courseId', breadcrumb: CourseBreadcrumb},
    { path: '/courses/:courseId/gradebook', breadcrumb: 'Gradebook'},
    { path: '/courses/:courseId/assignments/:assignmentId', breadcrumb: AssignmentBreadcrumb},
]
```

but for more detail pages such as submission feedback as well as createNCAG create CAG and update Assignment,![image](https://github.com/makeopensource/devU/assets/98787503/448697f2-ebd7-4336-9ad6-e0e69f0be925) it currently only show navBar until Assignment or Submission instead of update or feedback .etc
I dont know if this is the best soluction but just give you an idea
```
{ path: '/courses/:courseId/assignments/:assignmentId/:path', breadcrumb: DynamicBreadcrumb},
    { path: '/courses/:courseId/assignments/:assignmentId/submissions/:submissionId/:path', breadcrumb: DynamicBreadcrumb},
```
add those path in to the coutes

Reply:
I added dynamic pathing for routes that come after course and assignment using your function, but I don't think we should have intermediary breadcrumbs for parts of the url like "Courses" and "Assignments". While it is nicer for visual clarity, they link to non-existent/deprecated pages and don't serve a purpose for assisting navigation.

Reply:
the reason is that there are several links like courses homepage and assignment, I believe during the Sprint3 grading, Alan mentioned he does not even know which page he's on so that's why I added course/submission/ assignment/ in just to distinguish the exact page the user is on